external-pipe-api/src/main/java/org/apache/iotdb/pipe/external/api/IExternalPipeSinkWriter.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.pipe.external.api;
+
+import java.io.IOException;
+
+/** Responsible for forwarding the operations to the sink. */
+public interface IExternalPipeSinkWriter extends AutoCloseable {
```

Comment:
Could we use the interface like Session?

Reply:
@qiaojialin  I can not catch your key point.
IExternalPipeSinkWriter's every implement object will be run in 1 separate thread.
Ext-pipe customer can run several IExternalPipeSinkWriter threads for 1 ext pipe sink.